src/BugsnagUnity/Native/Android/NativePayloadClassWrapper.cs

```diff
@@ -22,5 +23,183 @@ public void SetNativeString(string key, string value)
+            var value = NativePointer.Call<AndroidJavaObject>(key);
+            if (value == null)
+            {
+                return null;
+            }
+            else
+            {
+                return value.Call<bool>("booleanValue");
+            }
+        }
+
+        public void SetNativeBool(string key, bool? value)
+        {
+            if (value == null)
+            {
+                NativePointer.Call(key, null);
+            }
+            else
+            {
+                var javaBoolean = new AndroidJavaObject("java.lang.Boolean", (bool)value);
+                NativePointer.Call(key, javaBoolean);
+            }
+        }
+
+        public string[] GetNativeStringArray(string key)
+        {
+            var array = NativePointer.Call<AndroidJavaObject>(key);
+            if (array == null)
+            {
+                return null;
+            }
+            var arraysClass = new AndroidJavaClass("java.util.Arrays");
+
+            var list = arraysClass.CallStatic<AndroidJavaObject>("asList",array);
+
+            var iterator = list.Call<AndroidJavaObject>("iterator");
+
+            var returnList = new List<string>();
+
+            while (iterator.Call<bool>("hasNext"))
+            {
+                var next = iterator.Call<string>("next");
+                returnList.Add(next);
+            }
+
+            return returnList.ToArray();
+
+        }
+
+        public void SetNativeStringArray(string key, string[] value)
+        {
+
+            if (value == null)
+            {
+                NativePointer.Call(key, null);
+                return;
+            }
+
+            AndroidJavaClass arrayClass = new AndroidJavaClass("java.lang.reflect.Array");
+
+            AndroidJavaObject arrayObject = arrayClass.CallStatic<AndroidJavaObject>("newInstance", new AndroidJavaClass("java.lang.String"), value.Length);
+
+            for (int i = 0; i < value.Length; ++i)
+            {
+                arrayClass.CallStatic("set", arrayObject, i, new AndroidJavaObject("java.lang.String", value[i]));
+            }
+            NativePointer.Call(key, arrayObject);
+
+        }
+
+        public Dictionary<string, object> GetNativeDictionary(string key)
+        {
+            var map = NativePointer.Call<AndroidJavaObject>(key);
+
+            if (map != null)
+            {
+                var size = map.Call<int>("size");
+
+                if (size > 0)
+                {
+                    var keys = map.Call<AndroidJavaObject>("keySet");
+
+                    var iterator = keys.Call<AndroidJavaObject>("iterator");
+
+                    var dict = new Dictionary<string, object>();
+
+                    while (iterator.Call<bool>("hasNext"))
+                    {
+                        var next = iterator.Call<AndroidJavaObject>("next");
+                        var theKey = next.Call<string>("toString");
+                        var theValue = map.Call<AndroidJavaObject>("get",theKey);
```

Comment:
Would probably be worth adding a null check - some maps implementations on the JVM can hold null values

Reply:
Done